src/gl/shader.cpp

```diff
@@ -44,23 +44,19 @@ bool find_id(const std::string& program, const char* id) {
     return std::strstr(program.c_str(), id) != 0;
 }
 
-bool Shader::load(const std::string& _fragmentSrc, const std::string& _vertexSrc, bool verbose) {
+bool Shader::load(const std::string* _fragmentPath, const std::string& _fragmentSrc, const std::string* _vertexPath, const std::string& _vertexSrc, bool verbose) {
     std::chrono::time_point<std::chrono::steady_clock> start_time, end_time;
     start_time = std::chrono::steady_clock::now();
 
-    m_vertexShader = compileShader(_vertexSrc, GL_VERTEX_SHADER);
+    m_vertexShader = compileShader(_vertexPath, _vertexSrc, GL_VERTEX_SHADER);
```

Comment:
`#include` works as expected. If there are warnings or errors in an included file, the filename I print is the name of the top level frag or vert file. Line numbers refer to the preprocessed file after all `#include` statements have been expanded. If you want this fixed, it's a big deal and needs its own issue.